File: 猿人学/sec4-js高阶逆向课/lesson12-补环境导论/11题.js
setTimeout = function (){}
window = {}
window.addEventListener = '1'
document = {}

document.addEventListener = function (a, b, c) {
    b()
}
document.createElement = function () {
    return {
        firstChild: {
            href: 'https://www.python-spider.com/'
        }
    }
}

__jscode

function decode() {
    return document.cookie
}

File: 猿人学/sec4-js高阶逆向课/lesson12-补环境导论/11题.py
import re

import execjs
import requests

cookies = {
    'sessionid': 'v5vxp57gehv2vgmckgmh19io2dgpuzk9'
}
session = requests.session()
url = 'https://www.python-spider.com/challenge/11'
response = session.get(url, cookies=cookies)
jscode = re.match('<script>(.*?)</script>', response.text).group(1)


with open('11题.js', 'r', encoding='utf-8') as f:
    result = execjs.compile(f.read().replace("__jscode",jscode)).call('decode')
cookies['__jsl_clearance'] = str(result).split("=")[1].split(";")[0]
response = session.get(url, cookies=cookies)
print(response.text)




